homeassistant/components/dsmr_reader/config_flow.py

```diff
@@ -0,0 +1,41 @@
+"""Config flow to configure DSMR Reader."""
+from __future__ import annotations
+
+from collections.abc import Awaitable
+import logging
+from typing import Any
+
+from homeassistant.components.mqtt.const import DOMAIN as MQTT_DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.config_entry_flow import DiscoveryFlowHandler
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _async_has_devices(hass: HomeAssistant) -> bool:
+    """Verify MQTT availability."""
+    return MQTT_DOMAIN in hass.config_entries.async_domains()
```

Comment:
If MQTT is setup from deprecated platform setup this won't be true.

Reply:
From reading the mqtt component's code, it would seem like it would create a config entry from the configuration.yaml, which should then get picked up. But I've seen another alternative: checking for the presence of any MQTT data. So I'll replace it with `return DATA_MQTT_CONFIG in hass.data` instead. I'm starting to run out of options here though 😅

Reply:
`return DATA_MQTT_CONFIG in hass.data` is still an implementation detail of `mqtt` that could change.

You've got `mqtt` as a dep here so let's just leave it as `True`

Sorry for the back and forth

Reply:
No problem! It allowed me to learn some more about the internals of HA. I've reverted the MQTT check.